### PR TITLE
SSS-42 Refactor PatientController to use Spring Data

### PR DESCRIPTION
**Pull request by jblefkove-va**
https://vasdvp.atlassian.net/browse/SSS-42

This has some impact on Bulk FHIR: The relationship between the two patient entities is now reversed (`PatientSearchEntity` has a `PatientEntity`)

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/PatientSearchEntity.java [hunk 1]
```diff
@@ -27,10 +30,10 @@
   private String icn;
 
   @Column(name = "firstName")
-  private String firstName;
+  private String given;
 
   @Column(name = "lastName")
-  private String lastName;
+  private String family;
```
[the remark this answers is not in the document]
Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/PatientSearchEntity.java]: I preferred `firstName`/`lastName`, but these names are more amenable to the search methods (e.g. `findByFamilyAndGender`) in `PatientRepository`

[the remark this answers is not in the document]
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/PatientSearchEntity.java]: I guess the trade off is: _Where are firstName/lastName mapped to given/family_?
This puts the mapping down to the column. _Given_ and _family_ are concepts from the REST api to the PatientRepository.
If left with _firstName_, then the mapping would happen at the REST controller as it selects which search methods to use, e.g `findByLastNameAndGender`.

I'm not passionate about either, though I prefer the entity properties to match the DB column names. I can live with this if everyone else is happy.

**Comment by bschofield-va** [comment]
I'm good with this, but I'll defer approving it until we sort out the Kong path issues we're having at the movement to prevent accidental merging... I don't want to introduce changes until we stabilize.

**Comment by bschofield-va** [comment]
Sorry mate... gonna need you squeeze 1 more percent.
```
20:49:34  [INFO]  94%   275  5473  Total
20:49:34  [INFO]  95% Required
```

**Pull request merged**